lib/Doctrine/DBAL/Schema/Column.php

```diff
@@ -92,106 +88,60 @@ public function setOptions(array $options)
         return $this;
     }
 
-    /**
-     * @return Column
-     */
-    public function setType(Type $type)
+    public function setType(Type $type) : Column
     {
         $this->_type = $type;
 
         return $this;
     }
 
-    /**
-     * @param int|null $length
-     *
-     * @return Column
-     */
-    public function setLength($length)
+    public function setLength(?int $length) : Column
     {
-        if ($length !== null) {
-            $this->_length = (int) $length;
-        } else {
-            $this->_length = null;
-        }
+        $this->_length = $length;
 
         return $this;
     }
 
-    /**
-     * @param int $precision
-     *
-     * @return Column
-     */
-    public function setPrecision($precision)
+    public function setPrecision(?int $precision) : Column
     {
-        if (! is_numeric($precision)) {
-            $precision = 10; // defaults to 10 when no valid precision is given.
-        }
-
-        $this->_precision = (int) $precision;
+        // defaults to 10 when no valid precision is given.
+        $this->_precision = $precision ?? 10;
 
         return $this;
     }
 
-    /**
-     * @param int $scale
-     *
-     * @return Column
-     */
-    public function setScale($scale)
+    public function setScale(?int $scale) : Column
     {
-        if (! is_numeric($scale)) {
-            $scale = 0;
-        }
-
-        $this->_scale = (int) $scale;
+        $this->_scale = $scale ?? 0;
```

Comment:
Maybe should use `int $scale = 0` directly?

Reply:
Same as above. It'd rather try to remove this default entirely and make it non-nullable.